src/main/kotlin/kweb/Kweb.kt

```diff
@@ -385,14 +388,8 @@ class Kweb private constructor(
             val cachedIds = mutableListOf<Int>()
             for (msg in initialMessages) {
                 val deserialedMsg = Json.decodeFromString<Server2ClientMessage>(msg)
-                //val deserialedMsg = gson.fromJson<Server2ClientMessage>(msg)
-
-                //For some reason the final msg in initialMessages looks like this,
-                //{"yourId":"gkUd4k","debugToken":"1446aab757c06931","js":""}
-                //I'm not sure what the point of this message is, and I can't find what is sending it.
-                //But, it causes problems if we try to add that to the cache, so we just make sure the jsId isn't null
-                //This is a useful check anyway, because we do let Server2ClientMessages have null jsId's, and trying to add
-                //a function from one of those messages wouldn't work.
+
+                //We have a special case where some functions do not have jsId's. Trying to add one of those to the cache would cause problems.
                 if (deserialedMsg.jsId != null) {
```

Comment:
Should this throw an error if jsId is null, rather than fail silently?

Reply:
I don't think so. This isn't a fail situation. I am going through the functions called on page render, caching the ones that need to be cached, and then executing all of these functions. We have support for calling JS functions that do not get cached, so it isn't a failure if the jsId is null here. We just need to not cache the function, or we will get a failure.